   Compiling c2rust_out v0.0.0 (/home/c00834010/dev/crusts-benchmarks/2023-06-27/nimber_arithmetic)
error: default numeric fallback might occur
  --> src/main.rs:22:15
   |
22 |         m >>= 1;
   |               ^ help: consider adding suffix: `1_i32`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_numeric_fallback
   = note: requested on the command line with `-D clippy::default-numeric-fallback`

error: unsafe block missing a safety comment
  --> src/main.rs:81:5
   |
81 |     unsafe {
   |     ^^^^^^^^
   |
   = help: consider adding a safety comment on the preceding line
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#undocumented_unsafe_blocks
   = note: requested on the command line with `-D clippy::undocumented-unsafe-blocks`

error: used `expect()` on an `Option` value
  --> src/main.rs:86:32
   |
86 |                 print!("{:3}", func.expect("non-null function pointer")(a_1, b));
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: if this value is `None`, it will panic
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expect_used
   = note: requested on the command line with `-D clippy::expect-used`

error: default numeric fallback might occur
   --> src/main.rs:113:12
    |
113 |     return 0;
    |            ^ help: consider adding suffix: `0_i32`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_numeric_fallback

error: could not compile `c2rust_out` (bin "main") due to 4 previous errors
